### PR TITLE
Add missing fields to fullData

### PR DESCRIPTION
**Pull request by etpinard**
covers a few items in https://github.com/plotly/plotly.js/issues/2834

----

Does not yet address pie `marker.colors` and violin `span` and `bandwidth` (see https://github.com/plotly/plotly.js/issues/2834#issuecomment-408198993 for more on that topic). 

cc @alexcjohnson

### CODE REVIEW

File: src/traces/box/plot.js [hunk 1]
```diff
@@ -304,7 +304,7 @@ function plotBoxMean(sel, axes, trace, t) {
 
     var paths = sel.selectAll('path.mean').data((
         (trace.type === 'box' && trace.boxmean) ||
-        (trace.type === 'violin' && trace.box && trace.meanline)
+        (trace.type === 'violin' && (trace.box || {}).visible && (trace.meanline || {}).visible)
```
Comment (alexcjohnson) [inline src/traces/box/plot.js]: can't this just be `trace.box.visible && trace.meanline.visible` given the changes below?

**Comment by etpinard** [comment]
Editor devs, is adding violins to the chart editor a high priority?

If so, I'm thinking of simply mutating the computed `span` and `bandwidth` default values back into  `gd._fullData` **and** `gd.data`  similar to how we currently handle histogram bins and contour levels.

**Comment by etpinard** [comment]
Ooops, https://github.com/plotly/plotly.js/pull/2850#issuecomment-408975034 was meant for https://github.com/plotly/plotly.js/issues/2834

**Comment by etpinard** [comment]
@alexcjohnson this PR isn't enough to cover all the items in https://github.com/plotly/plotly.js/issues/2834, but it does fix a few things. It would be nice to release this in next week's `v1.41.0`.

Tagging as reviewable.

**Comment by alexcjohnson** [comment]
Yep, this is great, no need to wait on the stragglers before merging this. 💃

**Pull request merged**